### PR TITLE
feat(order): add order module

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## What is the new behavior?
- Adds order `get` actions
- Adds order `list` actions

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

Import any order related things from `@daffodil/order` instead of `@daffodil/checkout`.

## Other information

### CODE REVIEW

**Comment by lderrickable** [comment]
I shouldn't have approved the order driver interface PR you put out. That code and this code should go in a new `@daffodil/order` package. Checkout will be the intersection between cart and order.

**Comment by griest024** [comment]
@lderrickable I have created the order package and moved everything over. There are now breaking changes in this PR

**Comment by lderrickable** [comment]
don't move deprecated stuff to the order package. Kind of defeats the purpose of deprecating it. Like the order containers and the place order actions don't need to move, since they're going to be deleted anyway.

**Comment by griest024** [comment]
@lderrickable I'll just put the new actions and the models in the order package then

**Comment by griest024** [comment]
@lderrickable made those changes. I'll move/deprecate the other parts of order in their specific PRs. LMK what you think

File: libs/order/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "@daffodil/core": "0.0.0-PLACEHOLDER"
+  },
+  "devDependencies": {
+    "@daffodil/core": "0.0.0-PLACEHOLDER",
+    "@daffodil/cart": "0.0.0-PLACEHOLDER",
+    "@daffodil/geography": "0.0.0-PLACEHOLDER",
+    "@daffodil/driver": "0.0.0-PLACEHOLDER"
+  },
+  "optionalDependencies": {
+    "@daffodil/driver": "0.0.0-PLACEHOLDER"
```
Comment (lderrickable) [inline libs/order/package.json:32]: I don't think this dependency is needed. Is it?

Comment (griest024) [reply libs/order/package.json]: no, I don't think it is. changed

**Comment by lderrickable** [comment]
lgtm

**Comment by griest024** [comment]
@lderrickable the only thing that happened since your approval is a rebase.

File: libs/order/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# @daffodil/order
```
Comment (damienwebdev) [inline libs/order/README.md:1]: This PR is pretty big. Can you truncate the changes to checkout/demo and do those in a separate PR?

I'd like this PR to only reflect the addition of `@daffodil/order` and some models/actions. The other changes can happen atomically in smaller commits & PRs.

Comment (griest024) [reply libs/order/README.md:1]: I have removed demo/checkout changes from this PR. Note that only the actions and driver interface actually contain functional changes from the old order module in the checkout package.

**Comment by damienwebdev** [comment]
@griest024. I think this is code-complete. Could also parse out the diffs into individual chunks for the features they represent?

**Comment by griest024** [comment]
@damienwebdev do you want individual PRs for each feature?

**Comment by damienwebdev** [comment]
@griest024 That's not quite necessary, as long as they follow the contributing guidelines, and have the PR # attached to them, that's sufficient.

**Comment by griest024** [comment]
@damienwebdev I think I've done it. LMK if its good to go or not.
